libsonata,py-libsonata: update to v0.1.26

mgeplf opened the pull request.

mgeplf committed:
libsonata,py-libsonata: update to v0.1.26

mgeplf commented:
Not sure I did this right; do I edit both of these files when something has been upstreamed?

jorblancoa commented on var/spack/repos/builtin/packages/py-libsonata/package.py:
```diff
@@ -16,6 +16,7 @@ class PyLibsonata(PythonPackage):
     maintainers("tristan0x")
 
     version("master", branch="master")
+    version("0.1.25", sha256="b653cbefbc511fe24ccf5cce7d80253954ec280800af5fd33700f0faea93fd4c")
```
```suggestion
    version("0.1.26", sha256="b653cbefbc511fe24ccf5cce7d80253954ec280800af5fd33700f0faea93fd4c")
```

mgeplf committed:
Update var/spack/repos/builtin/packages/py-libsonata/package.py

Co-authored-by: Jorge Blanco Alonso <41900536+jorblancoa@users.noreply.github.com>

mgeplf commented:
Looks like we need an update of bluecellulab; I will make a PR.

mgeplf committed:
py-bluecellulab,py-bluepysnap: bump versions

mgeplf commented:
@anilbey can you look at the updated py-bluecellulab to make sure the versions look ok?

mgeplf committed:
upper bound for old bluecellulab

anilbey commented:
> @anilbey can you look at the updated py-bluecellulab to make sure the versions look ok?

All good, thanks @mgeplf

mgeplf committed:
set bounds properly for bluecellulab

mgeplf committed:
bound bluepysnap

matz-e commented on bluebrain/repo-bluebrain/packages/py-bluepysnap/package.py:
```diff
@@ -17,17 +17,21 @@ class PyBluepysnap(PythonPackage):
+
     depends_on("py-libsonata@0.1.21:", type=("build", "run"))
-    depends_on("py-libsonata@0.1.24:", when="@2", type=("build", "run"))
+    depends_on("py-libsonata@0.1.24:", when="@2:2", type=("build", "run"))
+    depends_on("py-libsonata@0.1.26:", when="@3:3", type=("build", "run"))
```
```suggestion
    depends_on("py-libsonata@0.1.24:", when="@2", type=("build", "run"))
    depends_on("py-libsonata@0.1.26:", when="@3", type=("build", "run"))
```

mgeplf commented:
Looks like `bluecellulab` requires pydantic>2:
`bluecellulab 2.6.6 has requirement pydantic<3.0.0,>=2.5.2, but you have pydantic 1.10.9.`
Looking at upstream, they don't have it yet.  IIRC, pydantic requires rust and I don't have the time to update all that at the moment.

mgeplf committed:
py-annotated-types,py-pydantic: added/updated

mgeplf committed:
py-hatchling,py-pydantic-core,py-pydantic,rust-bootstrap,rust: update or add from upstream

mgeplf committed:
set py-cellulab pydantic bounds

matz-e commented on var/spack/repos/builtin/packages/py-annotated-types/package.py:
From upstream?

Note to self: create ticket about upstreaming

mgeplf replied:
No, I did that one myself.

mgeplf committed:
use newer bluecellulab that has fewer restrictions on h5py

mgeplf committed:
bound py-emodel-generalisation based on its setup.py

https://github.com/BlueBrain/emodel-generalisation/blob/0d2275a6fe69b0fa7850b4239671fd6425e2e524/setup.py#L47C26-L47C33

mgeplf committed:
update style

mgeplf committed:
rustfmt now comes from `rust+dev`

mgeplf committed:
add license

mgeplf committed:
more style

mgeplf committed:
I will learn to read sometime
